src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/FindUnwantedIslandsTask.cpp

```diff
@@ -52,9 +51,8 @@ void FindUnwantedIslandsTask::run() {
     int leftNucleotide = searchArea.startPos;
     /**
      * Index of the right nucleotide in the searching area.
-     * @settings.overlapLength.maxValue is the area extending deep into the amplified fragment.
      */
-    const int rightNucleotide = searchArea.endPos() + possibleOverlap;
+    const int rightNucleotide = searchArea.endPos() - 1;
```

Comment:
Since `endPos()` returns the end non-inclusive, I changed the index to match the name and further logic. So I also changed the logging in `FindUnwantedIslandsTask::regionToString`